packages/navie/src/commands/record-command.ts

```diff
@@ -0,0 +1,141 @@
+import { z } from 'zod';
+
+import type Command from '../command';
+import type { CommandRequest } from '../command';
+import CompletionService, { ModelType } from '../services/completion-service';
+import LookupContextService from '../services/lookup-context-service';
+import VectorTermsService from '../services/vector-terms-service';
+import { ContextV2 } from '../context';
+import { ExplainOptions } from './explain-command';
+import Message from '../message';
+
+const RelevantTest = z
+  .object({
+    name: z.string().optional().describe('The name of the test case, if known'),
+    path: z.string().describe('The path to the test file'),
+    language: z
+      .enum(['ruby', 'python', 'java', 'javascript', 'other'])
+      .describe('The programming language of the test file'),
+    framework: z.string().optional().describe('The test framework used'),
+  })
+  .optional()
+  .describe('An object containing the most relevant test to the requested behavior');
+
+export default class RecordCommand implements Command {
+  constructor(
+    private readonly options: ExplainOptions,
+    private readonly completionService: CompletionService,
+    private readonly lookupContextService: LookupContextService,
+    private readonly vectorTermsService: VectorTermsService
+  ) {}
+
+  getTestSnippets(vectorTerms: string[], tokenLimit: number): Promise<ContextV2.FileContextItem[]> {
+    return this.lookupContextService
+      .lookupContext(vectorTerms, tokenLimit, {
+        include: ['test', 'spec'],
+      })
+      .then((contextItems) =>
+        contextItems.filter(
+          (item): item is ContextV2.FileContextItem =>
+            item.type === ContextV2.ContextItemType.CodeSnippet
+        )
+      );
+  }
+
+  async getMostRelevantTest(
+    userRequest: string,
+    fileContextItems: ContextV2.FileContextItem[]
+  ): Promise<z.infer<typeof RelevantTest>> {
+    const context = fileContextItems.map(
+      ({ content, location }) =>
+        `<code-snippet location="${location}">\n${content}\n</code-snippet>`
+    );
+    const prompt = `Given the following code snippets, identify the single most relevant test to the user request.
+<code-snippets>
+${context.join('\n')}
+</code-snippets>`;
+
+    const response = await this.completionService.json(
+      [
+        {
+          role: 'system',
+          content: prompt,
+        },
+        {
+          role: 'user',
+          content: userRequest,
+        },
+      ],
+      RelevantTest,
+      { model: ModelType.Full }
```

Comment:
JSON completions don't work with the current `gpt-4o` pointer. Only the most recent model supports strict structured output in this manner. Maybe this can just use mini.

Reply:
TBH we have been getting by just fine with extracting code from fences and interpreting the result.